### PR TITLE
KAFKA-17609:[4/4]Convert system tests to kraft part 4

### PR DESCRIPTION
**Pull request by bbejeck**
1. [Streams smoke test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=aeec5685-ef5e-3652-99be-57e42f13889a)
2. [Static membership test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=3e865efc-f391-3b9a-b8d7-785d8f1edff9)
3. [Streams upgrade test]( https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=66df74fd-295d-3861-ba75-8d16626ad6bb)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
Links to passing tests

[Streams smoke test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=aeec5685-ef5e-3652-99be-57e42f13889a)
[Static membership test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=3e865efc-f391-3b9a-b8d7-785d8f1edff9)
[Streams upgrade test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=66df74fd-295d-3861-ba75-8d16626ad6bb)

File: tests/kafkatest/tests/streams/streams_smoke_test.py [hunk 1]
```diff
@@ -49,8 +49,8 @@ def __init__(self, test_context):
     @cluster(num_nodes=8)
     @matrix(processing_guarantee=['exactly_once_v2', 'at_least_once'],
             crash=[True, False],
-            metadata_quorum=quorum.all_non_upgrade)
-    def test_streams(self, processing_guarantee, crash, metadata_quorum=quorum.zk):
+            metadata_quorum=[quorum.combined_kraft])
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_smoke_test.py:52]: Can we remove `metadata_quorum` as parameter entirly ?

File: tests/kafkatest/tests/streams/streams_upgrade_test.py [hunk 2]
```diff
@@ -111,102 +105,10 @@ def perform_broker_upgrade(self, to_version):
...
-        processor.node.account.ssh_capture("grep SMOKE-TEST-CLIENT-CLOSED %s" % processor.STDOUT_FILE, allow_fail=False)
-
-    @cluster(num_nodes=6)
-    @matrix(from_version=metadata_1_versions)
-    @matrix(from_version=metadata_2_versions)
-    @matrix(from_version=fk_join_versions)
-    def test_rolling_upgrade_with_2_bounces(self, from_version):
+    @matrix(from_version=metadata_2_versions, metadata_quorum=[quorum.combined_kraft])
+    @matrix(from_version=fk_join_versions, metadata_quorum=[quorum.combined_kraft])
+    def test_rolling_upgrade_with_2_bounces(self, from_version, metadata_quorum):
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_upgrade_test.py:111]: Why are we adding `metadata_quorum` parameter? Seems unncessary?

File: tests/kafkatest/tests/streams/streams_upgrade_test.py [hunk 3]
```diff
@@ -245,7 +147,8 @@ def test_rolling_upgrade_with_2_bounces(self, from_version):
         self.stop_and_await()
 
     @cluster(num_nodes=6)
-    def test_version_probing_upgrade(self):
+    @matrix(metadata_quorum=[quorum.combined_kraft])
+    def test_version_probing_upgrade(self, metadata_quorum):
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_upgrade_test.py:151]: as above

File: tests/kafkatest/tests/streams/streams_upgrade_test.py [hunk 4]
```diff
@@ -272,8 +175,8 @@ def test_version_probing_upgrade(self):
         self.stop_and_await()
 
     @cluster(num_nodes=6)
-    @matrix(from_version=[str(LATEST_3_2), str(DEV_VERSION)],  upgrade=[True, False])
-    def test_upgrade_downgrade_state_updater(self, from_version, upgrade):
+    @matrix(from_version=[str(LATEST_3_2), str(DEV_VERSION)],  upgrade=[True, False], metadata_quorum=[quorum.combined_kraft])
+    def test_upgrade_downgrade_state_updater(self, from_version, upgrade, metadata_quorum):
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_upgrade_test.py:179]: as above

File: tests/kafkatest/tests/streams/streams_upgrade_test.py (hunk 2 above)
Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_upgrade_test.py:111]: Required to indicate to use KRaft for metadata quorum.  Failing to do so without a `@matrix` annotation results in this error as the test will default to ZK without it.
```File "/opt/kafka-dev/tests/kafkatest/services/kafka/kafka.py", line 541, in security_config
    zk_sasl=self.zk.zk_sasl if self.quorum_info.using_zk else False, zk_tls=self.zk_client_secure,
AttributeError: 'NoneType' object has no attribute 'zk_sasl'
```

File: tests/kafkatest/tests/streams/streams_smoke_test.py (hunk 1 above)
Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_smoke_test.py:52]: Required to indicate to use KRaft for metadata quorum.  Failing to do so without a `@matrix` annotation results in this error as the test will default to ZK without it.
```File "/opt/kafka-dev/tests/kafkatest/services/kafka/kafka.py", line 541, in security_config
    zk_sasl=self.zk.zk_sasl if self.quorum_info.using_zk else False, zk_tls=self.zk_client_secure,
AttributeError: 'NoneType' object has no attribute 'zk_sasl'
```

**Review by mjsax — APPROVED** [review]
You answered the open question on the other PRs. LGTM.

**Comment by bbejeck** [comment]
[Link for passing tests](https://semaphore.ci.confluent.io/jobs/d23b0787-df34-4b40-9500-dd9eeb13eea0/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976) in this PR

zk_to_kraft_pr_conversion_results

**Comment by bbejeck** [comment]
Merged #17328 into trunk

**Pull request merged**